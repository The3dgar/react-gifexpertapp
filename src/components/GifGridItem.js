import React from "react";

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} loading="lazy"></img>
      <p>{title}</p>
    </div>
  );
};
