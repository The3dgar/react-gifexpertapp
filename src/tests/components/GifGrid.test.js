import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("pruebas en <GifGrid></GifGrid>", () => {
  // Para saber cuantas veces fue llamada
  const category = "Friends";
  // let wrapper = shallow(<GifGrid category={category}></GifGrid>)

  beforeEach(() => {
    jest.clearAllMocks();
    // wrapper = shallow(<GifGrid category={category}></GifGrid>)
  });

  test("debe de mostrar correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category}></GifGrid>);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cosa/cosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "ABC",
        url: "https://localhost/cosa/cosa.jpg",
        title: "Cualquier cosa",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category}></GifGrid>);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
