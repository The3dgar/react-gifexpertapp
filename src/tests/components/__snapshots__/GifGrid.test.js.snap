// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid></GifGrid> debe de mostrar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Friends
  </h3>
  <p
    className="animate__animated animate_flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid></GifGrid> debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Friends
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cosa/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cosa/cosa.jpg"
    />
  </div>
</Fragment>
`;
